may-2018-reenabled-opcodes.md

```diff
@@ -0,0 +1,428 @@
+
+Impact of successful execution:
+* stack memory use is constant (slight reduction by `len(n)`)
+* number of elements on stack is constant
+
+Unit tests:
+* `OP_0 0 OP_SPLIT -> OP_0 OP_0`. Execution of OP_SPLIT on empty array results in two empty arrays.
+* `x 0 OP_SPLIT -> OP_0 x`
+* `x len(x) OP_SPLIT -> x OP_0`
+* `x (len(x) + 1) OP_SPLIT -> FAIL`
+
+## Bitwise logic
+
+The bitwise logic operators expect 'byte sequence' operands. The operands must be the same length. 
+* In the case of 'byte sequence' operands `OP_CAT` can be used to pad a shorter byte sequence to an appropriate length.
+* In the case of 'byte sequence' operands where the length of operands is not known until runtime a sequence of 0x00 bytes 
+  (for use with `OP_CAT`) can be produced using `OP_0 n OP_NUM2BIN`
+* In the case of numeric value operands `x n OP_NUM2BIN` can be used to pad a numeric value to length `n` whilst preserving 
+  the sign bit.
+
+### OP_AND
+
+    Opcode (decimal): 132
+    Opcode (hex): 0x84
+
+Boolean *and* between each bit in the operands.
+
+	x1 x2 OP_AND -> out
+
+Notes:
+* where `len(x1) == 0 == len(x2)` the output will be an empty array.
+
+The operator must fail if:
+1. `len(x1) != len(x2)`. The two operands must be the same size.
+
+Impact of successful execution:
+* stack memory use reduced by `len(x1)`
+* number of elements on stack is reduced by one
+
+Unit tests:
+
+1. `x1 x2 OP_AND -> failure`, where `len(x1) != len(x2)`. The two operands must be the same size.
+2. `x1 x2 OP_AND -> x1 & x2`. Check valid results.
+
+### OP_OR
+
+    Opcode (decimal): 133
+    Opcode (hex): 0x85
+
+Boolean *or* between each bit in the operands.
+
+	x1 x2 OP_OR -> out
+	
+The operator must fail if:
+1. `len(x1) != len(x2)`. The two operands must be the same size.
+
+Impact of successful execution:
+* stack memory use reduced by `len(x1)`
+* number of elements on stack is reduced by one
+
+Unit tests:
+1. `x1 x2 OP_OR -> failure`, where `len(x1) != len(x2)`. The two operands must be the same size.
+2. `x1 x2 OP_OR -> x1 | x2`. Check valid results.
+
+### OP_XOR
+
+    Opcode (decimal): 134
+    Opcode (hex): 0x86
+
+Boolean *xor* between each bit in the operands.
+
+	x1 x2 OP_XOR -> out
+	
+The operator must fail if:
+1. `len(x1) != len(x2)`. The two operands must be the same size.
+
+Impact of successful execution:
+* stack memory use reduced by `len(x1)`
+* number of elements on stack is reduced by one
+
+Unit tests:
+1. `x1 x2 OP_XOR -> failure`, where `len(x1) != len(x2)`. The two operands must be the same size.
+2. `x1 x2 OP_XOR -> x1 xor x2`. Check valid results.
+    
+## Arithmetic
+
+#### Note about canonical form and floor division
+
+Operands for all arithmetic operations are assumed to be numeric values and must be in canonical form.  
+See [data types](#data-types) for more information.
+
+**Floor division**
+
+Note: that when considering integer division and modulo operations with negative operands the rules applied in the C language and most
```

Comment:
As a grammar thing, the word "that" should not be here, or else the colon should be removed.

Also, there should be a comma after "operands".

Reply:
done, thanks